default.nix

```diff
@@ -1,22 +1,33 @@
 let
-  config = {
-    packageOverrides = super: rec {
+  thisOverlay = self: super: {
+    my-R = self.rWrapper.override {
+      packages = with self.rPackages; [ dplyr ];
+    };
 
-      haskellPackages = super.haskellPackages.override {
-        overrides = haskellPackagesNew: haskellPackagesOld: {
-          nix-mwe = haskellPackagesNew.callPackage ./cabal.nix {
-            R = super.rWrapper.override {
-              packages = with super.rPackages; [ dplyr ];
-            };
-          };
-        };
-      };
+    haskellPackages = super.haskellPackages.override {
+      overrides = hself: hsuper: {
+        nix-mwe = self.haskell.lib.overrideCabal (hself.callCabal2nix "nix-mwe" ./. {}) (oldAttrs: {
+          buildTools = oldAttrs.buildTools or [] ++ [ self.makeWrapper ];
+          postInstall = oldAttrs.postInstall or "" + ''
+            wrapProgram $out/bin/nix-mwe --prefix 'PATH' ':' "${self.lib.getBin self.my-R}/bin"
+          '';
+        });
 
+        inline-r = self.haskell.lib.overrideCabal hsuper.inline-r (oldAttrs: {
+          buildDepends = (oldAttrs.buildDepends or []) ++ [ self.R ];
+        });
```

Comment:
In my opinion, the `inline-r` nix packaging is slightly broken.

`inline-r` appears to require all dependent libraries to have access to an R repl, but this isn't specified in nix anywhere.

We need to add `R` to `buildDepends`, so that it is added to `propagatedBuildInputs`, so that `nix-mwe` also gets access to `R`.